### PR TITLE
Log info when refresh update is finished

### LINKED ISSUE #1376: Refresh status in spice runtime output should show when its done, and show the number of rows refreshed
**Describe the bug**
Spice runtime will show that its refreshing data into an accelerator, but it'll never indicate when its finished.  Example:

image

**To Reproduce**
Load the sample S3 data using the spice runtime.  Follow the README to complete the steps.

Observe that it shows its refreshing data, but never states when its completed (see screenshot).

**Expected behavior**
A log message indicating that its done loading, and displaying the number of rows that were loaded during the refresh.

### PR DESCRIPTION
**Pull request by y-f-u**
This adds affirmation when dataset full/incremental append refresh finishes.

The size is memory size of arrow batch record.

For appending dataset, first `Loaded` is the full refresh, second `Loaded` is the incremental one with no new data.

```
2024-05-21T12:19:34.288411Z  INFO runtime::accelerated_table::refresh: Loading data for dataset taxi_trips
2024-05-21T12:19:40.483899Z  INFO runtime::accelerated_table::refresh: Loaded 2964624 rows (421.71 MiB) for dataset taxi_trips in 6 seconds.
2024-05-21T12:19:44.293206Z  INFO runtime::accelerated_table::refresh: Loading data for dataset taxi_trips
2024-05-21T12:19:50.001700Z  INFO runtime::accelerated_table::refresh: Loaded 0 rows for dataset taxi_trips in 5 seconds.
```

### CODE REVIEW

**Review by ewgenius — DISMISSED** [review]

File: crates/runtime/src/accelerated_table/refresh.rs [hunk 1]
```diff
@@ -119,32 +120,37 @@ impl Refresher {
...
+                    if data_update.data.is_empty()
+                        || data_update
+                            .data
+                            .first()
+                            .map_or(false, |x| x.columns().is_empty())
+                    {
+                        if let Some(start_time) = start_time {
+                            if let Ok(elapse) = SystemTime::now()
+                                .duration_since(start_time)
+                                .map(|x| x.as_secs())
```
Comment (lukekim) [inline crates/runtime/src/accelerated_table/refresh.rs]: Does Rust not have a good time formatter like Go where it picks the best granularity?

File: crates/runtime/src/accelerated_table/refresh.rs [hunk 2]
```diff
@@ -154,6 +160,30 @@ impl Refresher {
...
                             } else {
+                                if let Some(start_time) = start_time {
+                                    let num_rows = data_update
+                                        .clone()
+                                        .data
+                                        .into_iter()
+                                        .map(|x| x.num_rows())
+                                        .sum::<usize>();
+
+                                    let memory_size = human_readable_bytes(
```
Comment (lukekim) [inline crates/runtime/src/accelerated_table/refresh.rs:171]: We will need to friendly format memory size.

File: crates/runtime/src/accelerated_table/refresh.rs [hunk 3]
```diff
@@ -478,6 +519,25 @@ pub(crate) fn get_timestamp(time: SystemTime) -> u128 {
         .as_nanos()
 }
 
+#[allow(clippy::cast_precision_loss)]
+#[allow(clippy::cast_sign_loss)]
+#[allow(clippy::cast_possible_truncation)]
+#[allow(clippy::cast_possible_wrap)]
+pub(crate) fn human_readable_bytes(num: usize) -> String {
```
Comment (lukekim) [inline crates/runtime/src/accelerated_table/refresh.rs]: Move to a util path. Is there really not a library that does this well like Go?

**Review by lukekim — DISMISSED** [review]

File: crates/runtime/src/accelerated_table/refresh.rs [hunk 4]
```diff
@@ -478,6 +519,25 @@ pub(crate) fn get_timestamp(time: SystemTime) -> u128 {
...
+    let units = ["B", "kiB", "MiB", "GiB"];
```
Comment (lukekim) [inline crates/runtime/src/accelerated_table/refresh.rs]: Add unit tests.

**Review by sgrebnov — APPROVED** [review]

File: crates/runtime/src/accelerated_table/refresh.rs (hunk 1 above)
Comment (y-f-u) [reply crates/runtime/src/accelerated_table/refresh.rs]: used a humantime (which is already part of the crate dependencies)

**Pull request merged**